### PR TITLE
Add support images -a

### PR DESCRIPTION
**Pull request by jimmyxian**
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by chanwit** [comment]
@jimmyxian It seems we need some integration tests for this PR :)

**Comment by jimmyxian** [comment]
@chanwit Yeah..Forget to `git commit`. :)

File: cluster/swarm/cluster.go [hunk 1]
```diff
@@ -298,7 +298,7 @@ func (c *Cluster) RemoveImages(name string) ([]*dockerclient.ImageDelete, error)
 	errs := []string{}
 	var err error
 	for _, e := range c.engines {
-		for _, image := range e.Images() {
+		for _, image := range e.Images(true) {
```
Comment (chanwit) [inline cluster/swarm/cluster.go]: Correct me if I am wrong. The default value is `false` so why `true` is used here?

File: scheduler/node/node.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func NewNode(e *cluster.Engine) *Node {
 		Name:        e.Name,
 		Labels:      e.Labels,
 		Containers:  e.Containers(),
-		Images:      e.Images(),
+		Images:      e.Images(true),
```
Comment (chanwit) [inline scheduler/node/node.go]: Here is also.

File: cluster/engine.go [hunk 3]
```diff
@@ -185,7 +185,7 @@ func (e *Engine) RemoveImage(image *Image, name string) ([]*dockerclient.ImageDe
 
 // RefreshImages refreshes the list of images on the engine.
 func (e *Engine) RefreshImages() error {
-	images, err := e.client.ListImages(false)
+	images, err := e.client.ListImages(true)
```
Comment (chanwit) [inline cluster/engine.go]: I am not sure why we need to `ListImages(true)` instead of the old `false`?

**Comment by jimmyxian** [comment]
@chanwit https://github.com/docker/swarm/pull/1085#issuecomment-126522219
* `all=true` : get all images(including intermediate layers)
* `all=false` : get the images which have repotag

**Comment by chanwit** [comment]
@jimmyxian Thank for clarifying.
LGTM

**Comment by abronan** [comment]
LGTM

**Pull request merged**